### PR TITLE
Fix Property Details and Review pages

### PR DESCRIPTION
**Pull request by dimak1**
*Issue:*

- bcgov/entity/issues/23547
- bcgov/entity/issues/24009

*Description of changes:*
- Fix dynamic dropdown for Property Type
- Rearrange Property Details in Review page as per designs
- Fix Application and Registration Details pages
- Tech debt

Screenshot 2024-11-05 at 13 19 07

### Registration Details

Screenshot 2024-11-06 at 12 32 22

 
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

File: strr-web/interfaces/account-i.ts [hunk 1]
```diff
@@ -170,7 +170,7 @@ export interface CreateAccountFormStateI {
     listingDetails: { url: string }[]
     rentalUnitSpaceType: string
     isUnitOnPrincipalResidenceProperty: boolean | undefined
-    hostResidence: string | undefined
+    hostResidence: string | null
```
Comment (shaangill025) [inline strr-web/interfaces/account-i.ts:173]: This needs to be applied elsewhere as well.

**Review by avni-work — APPROVED** [review]

**Review by shaangill025 — APPROVED** [review]

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-270-bbcap2d6.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-270-bbcap2d6.web.app

**Pull request merged**